pkg/controller/replication_controller_test.go

```diff
@@ -450,3 +432,321 @@ func TestSortingActivePods(t *testing.T) {
 		}
 	}
 }
+
+// NewFakeRCExpectationsLookup creates a fake store for PodExpectations.
+func NewFakeRCExpectationsLookup(ttl time.Duration) (*RCExpectations, *util.FakeClock) {
+	fakeTime := time.Date(2009, time.November, 10, 23, 0, 0, 0, time.UTC)
+	fakeClock := &util.FakeClock{fakeTime}
+	ttlPolicy := &cache.TTLPolicy{ttl, fakeClock}
+	ttlStore := cache.NewFakeExpirationStore(
+		expKeyFunc, nil, ttlPolicy, fakeClock)
+	return &RCExpectations{ttlStore}, fakeClock
+}
+
+func TestRCExpectations(t *testing.T) {
+	ttl := 30 * time.Second
+	e, fakeClock := NewFakeRCExpectationsLookup(ttl)
+	adds, dels := 10, 30
+	rc := newReplicationController(nil, 1)
+
+	// RC fires off adds and deletes at apiserver, then sets expectations
```

Comment:
Yeah it's fine in a test, but worth mentioning that you're testing it that way for convenience only.